helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -103,25 +100,30 @@ protected synchronized boolean doRegister(Collection<DynamicMetric<?, ?>> dynami
    * @param description description of the MBean
    * @param dynamicMetrics the DynamicMetrics
    */
-  private void updateAttributtInfos(Collection<DynamicMetric<?, ?>> dynamicMetrics,
+  protected void updateAttributesInfo(Collection<DynamicMetric<?, ?>> dynamicMetrics,
       String description) {
-    _attributeMap.clear();
+    if (dynamicMetrics == null || dynamicMetrics.isEmpty()) {
```

Comment:
I think so, as well. The original code checks null, so I do it to avoid changing the logic. If you also agree on not checking null, I can remove the check.

Reply:
Check for null looks good to me. I mean "dynamicMetrics.isEmpty()" is not necessary.
This is a way to remove the metrics in the list.

Reply:
Changed.

Reply:
Using an empty collection to remove the metrics is kind of hacky way to me. Let's confirm:
1. Do we allow null or empty to delete metrics? The original code allows this.
2. If we don't allow null to delete metrics, we will do nothing if it is null. We allow empty collection to delete metrics.

Reply:
Or you can disallow it and add another method to clear the list. Both work for me : )